Sources/HipMobileUI/Pages/UserRatingPage.xaml.cs

```diff
@@ -14,14 +14,61 @@
 
 using PaderbornUniversity.SILab.Hip.Mobile.UI.Navigation;
 using PaderbornUniversity.SILab.Hip.Mobile.UI.ViewModels.Pages;
+using Xamarin.Forms;
 
 namespace PaderbornUniversity.SILab.Hip.Mobile.UI.Pages
 {
     public partial class UserRatingPage : IViewFor<UserRatingPageViewModel>
     {
+        private double width;
+        private double height;
+
         public UserRatingPage()
         {
             InitializeComponent();
         }
+
+        /// <summary>
+        /// Size changed, determine if we need to update the layout.
+        /// </summary>
+        /// <param name="width">The new width.</param>
+        /// <param name="height">The new height.</param>
+        protected override void OnSizeAllocated(double width, double height)
+        {
+            base.OnSizeAllocated(width, height);
+            if (width != this.width || height != this.height)
```

Comment:
At first, I wanted to note that this does not correspond to the way we implemented the device orientation change. But apparently we have multiple alternative implementations of this so this is fine for now and I will take care of making the code unified.

One thing you should do here though is using `Math.Abs()` for the comparison with a deviation value of `0.00001`.

Reply:
I have got this code from the xamarin website about handling different device orientations (https://developer.xamarin.com/guides/xamarin-forms/user-interface/layouts/device-orientation), so I think this should be the way we implement the different screen orientations.
Why should I use Math.Abs() with a deviation value? The screen size shouldn't be varying, right?

Reply:
Most likely not. It's just recommended by Resharper and I also recall Bastian talking about using Math.Abs(). If you want to know more about the reasoning behind using it, you could ask him.

And I have seen that page before; perhaps I will create a small bug for unifying this part of our code to correspond to the approach noted there.